tests/rados/test_objectstore_block.py

```diff
@@ -0,0 +1,302 @@
+import datetime
+import json
+import time
+
+from ceph import utils as ceph_utils
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    #CEPH-83571714
+    Capture the block stats of an object using ceph-objectstore-tool utility
+    Verify the decrease in number of blocks when the amount of data in an object
+    is reduced by overwrite
+    1. Create pool and write data to an object
+    2. Fetch the acting pg set for the object and corresponding osd nodes
+    3. Stop any one of the OSDs from the acting pg set using systemctl
+    4. Capture the object dump using ceph-objectstore-tool
+    5. Start the OSD which was previously stopped
+    6. Overwrite the data in the object with a smaller chunk
+    7. Stop the same OSD and fetch object dump using ceph-objectstore-tool
+    8. Verify the difference in block stats and decrease in the number of blocks
+    should be proportional to the decrease in data in the object
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    rhbuild = config.get("rhbuild")
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    pool_obj = PoolFunctions(node=cephadm)
+    client_node = ceph_cluster.get_nodes(role="client")[0]
+    installer_node = ceph_cluster.get_nodes(role="installer")[0]
+    iterations = config.get("write_iteration")
+    timeout = config.get("timeout", 30)
+
+    pool_name = config["pool_name"]
+    object_name = config.get("object_name", "obj-objectstore")
+
+    log.info(
+        "Running test case to verify changes in number "
+        "of blocks when data is written to an object and removed from an object"
+    )
+
+    try:
+        # create pool with given config
+        if config["create_pool"]:
+            rados_obj.create_pool(pool_name=pool_name)
+
+        # retrieve value of config parameter bluestore_min_alloc_size_hdd
+        bluestore_min_alloc_size, _ = installer_node.exec_command(
+            sudo=True,
+            cmd="cephadm shell -- ceph config get osd bluestore_min_alloc_size_hdd",
+        )
+        bluestore_min_alloc_size = int(bluestore_min_alloc_size)
+
+        # create single object and perform IOPs
+        pool_obj.do_rados_put(client=client_node, pool=pool_name, obj_name=object_name)
+
+        pool_obj.do_rados_append(
+            client=client_node,
+            pool=pool_name,
+            obj_name=object_name,
+            nobj=(iterations - 1),
+        )
+
+        osd_map = rados_obj.get_osd_map(pool=pool_name, obj=object_name)
+        acting_pg_set = osd_map["acting"]
+        log.info(f"Acting pg set for {object_name} in {pool_name}: {acting_pg_set}")
+        if not acting_pg_set:
+            log.error("Failed to retrieve acting pg set")
+            return 1
```

Comment:
Please Raise Exceptions in case of failures instead of return 1.

Reply:
Replaced all applicable return statements with raise exceptions